File: sub-web-api/src/main/java/org/github/web/module/sys/menu/controller/MenuController.java
package org.github.web.module.sys.menu.controller;

import java.util.List;
import org.github.base.entity.SysMenuEntity;
import org.github.spring.restful.Returnable;
import org.github.web.module.sys.menu.model.bo.QueryMenuListBO;
import org.github.web.module.sys.menu.service.IMenuService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;

@RequestMapping("/sys/menu/")
@Controller
public class MenuController {
  @Autowired
  private IMenuService menuService;

  @PostMapping("del")
  public Returnable delMenuList(@RequestBody List<Long> ids) {
    return menuService.delMenuList(ids);
  }

  @GetMapping("info/{menuId}")
  public Returnable queryMenuInfo(@PathVariable Long menuId) {
    return menuService.queryMenuInfo(menuId);
  }

  @PostMapping("list")
  public Returnable queryMenuList(@RequestBody QueryMenuListBO bo) {
    return menuService.queryMenuList(bo);
  }

  @PostMapping("select")
  public Returnable queryMenuTree() {
    return menuService.queryMenuTree();
  }

  @PostMapping({"save", "update"})
  public Returnable saveOrUpdate(@RequestBody SysMenuEntity bo) {
    return menuService.saveOrUpdate(bo);
  }
}

File: sub-module-web/src/main/java/org/github/controller/MenuController.kt
package org.github.controller

import cn.hutool.core.bean.BeanUtil.trimStrFields
import org.github.sso.module.sys.menu.dto.QueryMenuListBO
import org.github.sso.module.sys.menu.dto.SaveOrUpdateMenuBO
import org.github.web.module.sys.menu.service.IMenuService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RequestMapping("/sys/menu/")
@RestController
class MenuController {
  @Autowired
  private lateinit var menuService: IMenuService

  @PostMapping("del")
  fun delMenuList(@RequestBody ids: Array<Long>) = menuService.delMenuList(ids.toList())

  @GetMapping("info/{menuId}")
  fun queryMenuInfo(@PathVariable menuId: Long) = menuService.queryMenuInfo(menuId)

  @PostMapping("page")
  fun queryMenuPage(@RequestBody bo: QueryMenuListBO) = menuService.queryMenuPage(trimStrFields(bo))

  @PostMapping("select")
  fun queryMenuTree() = menuService.queryMenuTree()

  @PostMapping("save", "update")
  fun saveOrUpdate(@RequestBody bo: SaveOrUpdateMenuBO) = menuService.saveOrUpdate(trimStrFields(bo))
}
